addons/web_editor/static/src/js/wysiwyg/widgets/link_popover_widget.js

```diff
@@ -0,0 +1,182 @@
+            // Preview pages from current website only. External website will
+            // most of the time raise a CORS error. To avoid that error, we
+            // would need to fetch the page through the server (s2s), involving
+            // enduser fetching problematic pages such as illicit content.
+            this.$('.js_preview_favicon img').attr({
+                'src': 'https://www.google.com/s2/favicons?sz=16&domain=' + url
+            }).removeClass('d-none');
+            this.$('.js_preview_favicon .fa').addClass('d-none');
+            this.$('.js_url_link').text(url);
+        } else {
+            $.get(this.target.href).then(function (content) {
+                var parser = new window.DOMParser();
+                var doc = parser.parseFromString(content, "text/html");
+
+                // Get
+                const favicon = doc.querySelector("link[rel~='icon']");
+                const ogSiteName = doc.querySelector("[property='og:site_name']");
+                const ogTitle = doc.querySelector("[property='og:title']");
+                const title = doc.querySelector("title");
+                const ogDescription = doc.querySelector("[property='og:description']");
+                const ogImage = doc.querySelector("[property='og:image']");
+                if (ogSiteName) {
+                    site += ' | ' + ogSiteName.getAttribute('content');
+                }
+
+                // Set
+                if (favicon) {
+                    self.$('.js_preview_favicon img').attr({'src': favicon.href}).removeClass('d-none');
+                    self.$('.js_preview_favicon .fa').addClass('d-none');
+                }
+                self.$('.js_preview_site').text(site);
+                if (ogTitle) {
+                    self.$('.js_url_link').text(ogTitle.getAttribute('content'));
+                } else if (title) {
+                    self.$('.js_url_link').text(title.text.trim());
+                }
+                if (ogDescription) {
+                    self.$('.js_preview_description').text(ogDescription.getAttribute('content')).removeClass('d-none');
+                }
+                if (ogImage) {
+                    self.$('.js_preview_image').attr({'src': ogImage.getAttribute('content')}).removeClass('d-none');
+                }
+            });
+        }
+    },
+    /**
+     * Resets the preview elements visibility. Particularly useful when changing
+     * the link url from an internal to an external one and vice versa.
+     *
+     * @private
+     */
+    _resetPreview() {
+        this.$('.js_preview_favicon img').addClass('d-none');
+        this.$('.js_preview_favicon .fa').removeClass('d-none');
+        this.$('.js_preview_description').addClass('d-none');
+        this.$('.js_preview_image').addClass('d-none');
+        this.$('.js_preview_site').text("");
+        // Useful for 404 internal URL
+        this.$('.js_url_link').text(this.$target.attr('href'));
```

Comment:
Since you search for those here and also during link update & stuff, I would save the search in a variable and use that variable. That would avoid repeating the selectors.